### ISSUE TITLE
Assembler misinterprets uppercase flags

### ISSUE THREAD

**Issue by borchers** [Dotneteer/kliveide#495]
The assembler does not recognize upper-case flags correctly. It still quietly compiles and runs the code, but the flag is not processed right. Here is an example:

`call Z, 1234` does not call a subroutine when the Z flag is set.
`call z, 1234` however works as expected, calling the subroutine when the Z flag is set.

This is also true, for example, for a conditional jump (`jr z` vs. `jr Z`).

Here is a complete code example . Change the "z" to "Z" to see the effect.

```
    .model Spectrum48
    .org #8000
    ld a, 42
    cp 42
    call z, Green       ; this works with "z" but not with "Z"
    ret

Green:
    ld a,4
    call 8859           ; set border color to green
    ret
```

**Comment by Dotneteer** [comment]
@borchers, Thanks for reporting this issue, I will soon fix it!

**Comment by Dotneteer** [comment]
Fixed in release [v0.12.4](https://github.com/Dotneteer/kliveide/releases/tag/v0.12.4).

**Comment by borchers** [comment]
Much appreciated, thank you Istvan!
